python/packages/azurefunctions/agent_framework_azurefunctions/_app.py

```diff
@@ -117,14 +126,17 @@ def my_orchestration(context):
         agents: Dictionary of agent name to AgentProtocol instance
         enable_health_check: Whether health check endpoint is enabled
         enable_http_endpoints: Whether HTTP endpoints are created for agents
+        enable_mcp_tool_endpoint: Whether MCP tool endpoints are created for agents
         max_poll_retries: Maximum polling attempts when waiting for responses
         poll_interval_seconds: Delay (seconds) between polling attempts
     """
 
     agents: dict[str, AgentProtocol]
     enable_health_check: bool
     enable_http_endpoints: bool
+    enable_mcp_tool_endpoint: bool
     agent_http_endpoint_flags: dict[str, bool]
+    agent_mcp_tool_flags: dict[str, bool]
```

Comment:
This is the 3rd dictionary we're adding to keep track of state (`agents`, `agent_http_endpoint_flags` and `agent_mcp_tool_flags` ). 
It might be cleaner to create a new class `AgentMetadata` or something and have it store each of these flag details and maintain a single dictionary `dict[str, AgentMetadata]` and use it everywhere. This way the code is cleaner and lesser chance of any sort of missed updates.